### PR TITLE
#802 created iterator of floats and ints

### PR DESCRIPTION
**Pull request by krzyk**
#802 
- created `IteratorOfFloats` and `IteratorOfInts` and updated the appropriate Iterables
- added tests for those new classes
- added todo to continue implementation of longs and shorts

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#832](https://github.com/yegor256/cactoos/issues/832) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#832](https://github.com/yegor256/cactoos/issues/832) is assigned to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman), here is [why](https://www.0crat.com/footprint/C63314D6Z/1272065005); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/test/java/org/cactoos/iterator/IteratorOfFloatsTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Test(expected = NoSuchElementException.class)
+    public void emptyIteratorThrowsException() {
+        new IteratorOfFloats().next();
+    }
+
+    @Test
+    public void nonEmptyIteratorDoesNotHaveNext() {
+        MatcherAssert.assertThat(
+            "Can't create non empty iterator",
```
Comment (proshin-roman) [inline src/test/java/org/cactoos/iterator/IteratorOfFloatsTest.java]: @krzyk here again: the failure message is not correct - it should explain the cause of assertion failing: an iterator with fetched items still has the next item which is wrong. Please, fix the message.

File: src/test/java/org/cactoos/iterator/IteratorOfFloatsTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            this.iteratorWithFetchedElements().hasNext(),
+            CoreMatchers.equalTo(false)
```
Comment (proshin-roman) [inline src/test/java/org/cactoos/iterator/IteratorOfFloatsTest.java]: @krzyk And here too: please, use `IsEqual`

File: src/test/java/org/cactoos/iterator/IteratorOfIntsTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * Tests for {@link IteratorOfInts}.
+ *
+ * <p>There is no thread-safety guarantee.</p>
+ *
+ * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
+ * @version $Id$
+ * @since 0.32
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class IteratorOfIntsTest {
```
Comment (proshin-roman) [inline src/test/java/org/cactoos/iterator/IteratorOfIntsTest.java]: @krzyk all comments written for `IteratorOfFloatsTest` can be applied for this class as well.

File: src/main/java/org/cactoos/iterable/IterableOfInts.java [hunk 4]
```diff
@@ -40,13 +40,8 @@
...
-        super(() -> {
-            final Collection<Integer> iterable =
-                new ArrayList<>(values.length);
-            for (final int value: values) {
-                iterable.add(value);
-            }
-            return iterable;
-        });
+        super(() -> ()
+            -> new UncheckedScalar<>(() -> new IteratorOfInts(values)).value()
```
Comment (proshin-roman) [inline src/main/java/org/cactoos/iterable/IterableOfInts.java]: @krzyk I might be wrong here (please, correct me if it's so), but using of `UncheckedScalar` looks redundant here, as the `super` ctor already wraps the argument into `UncheckedScalar`: https://github.com/yegor256/cactoos/blob/5ead4ac1184cfb35f848a8e8f1f181ac70a9270f/src/main/java/org/cactoos/iterable/IterableEnvelope.java#L54
Could you check please if that's correct and UncheckedScalar can be ommitted here?

Comment (krzyk) [reply src/main/java/org/cactoos/iterable/IterableOfInts.java]: @proshin-roman unchecked scalar is for `Iterator` interface (there are 3 lambdas here) so I need to wrap it at the bottom, wrapping in IterableEnvelope doesn't help here

**Comment by krzyk** [comment]
@proshin-roman I've updated the code, please take a look

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @proshin-roman[/z](https://www.0crat.com/u/proshin-roman), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#832](https://github.com/yegor256/cactoos/issues/832) is now out of scope

File: src/main/java/org/cactoos/iterable/IterableOfInts.java (hunk 4 above)
Comment (victornoel) [reply src/main/java/org/cactoos/iterable/IterableOfInts.java]: @krzyk why didn't you just write `() -> () -> new IteratorOfInts(values)`? I don't get what you wrote provides. The constructor of `IteratorOfInts` does not throw any exception...

Comment (victornoel) [reply src/main/java/org/cactoos/iterable/IterableOfInts.java]: @krzyk or `() -> new IterableOf<>(new IteratorOfInts(values))` if you prefer

Comment (krzyk) [reply src/main/java/org/cactoos/iterable/IterableOfInts.java]: @victornoel I didn't know I could (regarding the first of your comments - it wasn't possible, I didn't tried the second one)

Comment (victornoel) [reply src/main/java/org/cactoos/iterable/IterableOfInts.java]: @krzyk ok thanks, I will take that into account for solving #838 :)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @proshin-roman[/z](https://www.0crat.com/u/proshin-roman)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**